cns/restserver/internalapi.go

```diff
@@ -254,3 +262,60 @@ func (service *HTTPRestService) CreateOrUpdateNetworkContainerInternal(req cns.C
 	return returnCode
 
 }
+
+// Try to register node with DNC when CNS is started in managed DNC mode
+func RegisterNode(httpc *http.Client, httpRestService cns.HTTPService, dncEP, infraVnet, nodeID string) string {
+	logger.Printf("[Azure CNS] Registering node %s with Infrastructure Network: %s PrivateEndpoint: %s", nodeID, infraVnet, dncEP)
+
+	var (
+		numCPU              = runtime.NumCPU()
+		url                 = fmt.Sprintf(common.RegisterNodeURLFmt, dncEP, infraVnet, nodeID, dncApiVersion)
+		response            *http.Response
+		err                 = fmt.Errorf("")
+		body                bytes.Buffer
+		nodeRegisterRequest cns.NodeRegisterRequest
+		retMsg              string
+	)
+
+	nodeRegisterRequest.NumCPU = numCPU
+	supportedApis, msg := nmagentclient.GetNmAgentSupportedApis(httpc, "")
+
+	if msg != "" {
+		logger.Printf("[Azure CNS] Failed to retrieve SupportedApis from NMagent of node %s with Infrastructure Network: %s PrivateEndpoint: %s",
+			nodeID, infraVnet, dncEP)
+	}
+
+	nodeRegisterRequest.NmAgentSupportedApis = supportedApis
```

Comment:
Out of curiosity. Why does DNC need this information from nmagent when the node is registered?

Reply:
DNC needs this information to assign GRE key to a NC's primary IP Allocation. 2 conditions are required to be satisfied for DNC to start allocating GRE keys 1. NMAgent needs to have the support, 2. No other NC without greKey allocations to be present in the Node.